app/scripts/lib/setupSentry.js

```diff
@@ -146,31 +154,88 @@ export default function setupSentry({ release, getState }) {
       } else {
         return null;
       }
-      return breadcrumb;
+      return rewriteBreadcrumb(breadcrumb);
     },
   });
 
-  function rewriteReport(report) {
-    try {
-      // simplify certain complex error messages (e.g. Ethjs)
-      simplifyErrorMessages(report);
-      // modify report urls
-      rewriteReportUrls(report);
-      // append app state
-      if (getState) {
-        const appState = getState();
-        if (!report.extra) {
-          report.extra = {};
-        }
-        report.extra.appState = appState;
+  return Sentry;
+}
+
+function hideUrlIfNotInternal(_url) {
+  const re = /^(chrome-extension|moz-extension):\/\//u;
+  if (!_url.match(re)) {
+    return '';
+  }
+  return _url;
+}
+
+export function rewriteBreadcrumb(breadcrumb) {
+  /** @todo check URL allowlist, exclude URLs not in allowlist */
+
+  if (breadcrumb.data?.url) {
+    breadcrumb.data.url = hideUrlIfNotInternal(breadcrumb.data.url);
+  }
+  if (breadcrumb.data?.to) {
+    breadcrumb.data.to = hideUrlIfNotInternal(breadcrumb.data.to);
+  }
+  if (breadcrumb.data?.from) {
+    breadcrumb.data.from = hideUrlIfNotInternal(breadcrumb.data.from);
+  }
+  return breadcrumb;
+}
+
+export function rewriteReport(report, getState) {
```

Comment:
Nit: Ideally we'd have JSDoc comments for each of these functions, especially now that you've exported them.

Reply:
Added JSDoc comments in latest commit https://github.com/MetaMask/metamask-extension/pull/16780/commits/daaac950566477c5197089c509babada82f9cdaa